src/iam/jwt.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2018, Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Handling and verification of JWT claims."""
+
+import logging
+from functools import wraps
+
+from flask import abort, g, request
+from jose import jwt
+
+
+logger = logging.getLogger(__name__)
+
+
+def init_app(app):
+    """Add the jwt decoding middleware to the app."""
+    @app.before_request
+    def decode_jwt():
+        if 'Authorization' not in request.headers:
+            logger.debug("No JWT provided")
+            g.jwt_valid = False
+            g.jwt_claims = {'prj': {}}
+            return
+
+        auth = request.headers['Authorization']
+        if not auth.startswith('Bearer '):
+            g.jwt_valid = False
+            g.jwt_claims = {'prj': {}}
+            return
+
+        try:
+            _, token = auth.split(' ', 1)
+            g.jwt_claims = jwt.decode(
+                token, app.config['RSA_PUBLIC_KEY'], app.config['ALGORITHM'])
+            # JSON object names can only be strings. Map project ids to ints for
+            # easier handling
+            g.jwt_claims['prj'] = {
+                int(key): value
+                for key, value in g.jwt_claims['prj'].items()
+            }
+            g.jwt_valid = True
+            logger.debug(f"JWT claims accepted: {g.jwt_claims}")
+        except (jwt.JWTError, jwt.ExpiredSignatureError,
```

Comment:
If the above `auth.split` fails to unpack this will yield a `ValueError` or with my suggestion an `IndexError`. That should probably be caught here?

Reply:
This actually cannot be the case because of [the check on L39](https://github.com/DD-DeCaF/iam/pull/14/files#diff-546df41352a3e0c0653b4d85eaf38494R39) guarantees a space char in the value. That's not really obvious so I will add a comment.